src/client/opamSwitchCommand.ml

```diff
@@ -766,13 +769,26 @@ let guess_compiler_invariant ?repos rt strings =
         with Failure _ | Not_found ->
         try
           let v = OpamPackage.Version.of_string str in
+          let filter =
+            OpamPackage.Set.filter (fun nv ->
+                OpamPackage.Version.equal nv.version v)
+          in
           let candidates =
-            OpamPackage.Set.filter (fun nv -> nv.version = v)
-              compiler_packages
+            let candidates = filter compiler_packages in
+            if OpamPackage.Set.is_empty candidates then
+              let avoided_candidates = filter removed_compiler_packages in
+              if OpamPackage.Set.is_empty avoided_candidates then
+                raise Not_found
+              else
+                (OpamConsole.note
```

Comment:
Thinking about it, i feel like the note doesn't make much sense when only one version is available. It's like scolding users for being explicit. Adding a reftest for this would be nice too.
```suggestion
                (if not (OpamPackage.Set.is_singleton avoided_candidates) then
                   OpamConsole.note
```

Reply:
It just needs a rephrase to handle singleton. OR you are thinking about something else where there is only 1 package ?

Reply:
No what i meant is i don't think we should show a note at all when there is only one possible version